packages/grafana-ui/src/components/PieChart/PieChart.tsx

```diff
@@ -153,19 +136,51 @@ export const PieChart: FC<PieChartProps> = ({
   );
 };
 
+function useSliceHighlightState() {
+  const [highlightedTitle, setHighlightedTitle] = useState<string>();
+  const { eventBus } = usePanelContext();
+
+  useEffect(() => {
+    if (!eventBus) {
+      return;
+    }
+
+    const setHighlightedSlice = (event: DataHoverEvent) => {
+      if (eventBus.isOwnEvent(event)) {
+        setHighlightedTitle(event.payload.dataId);
+      }
+    };
+
+    const resetHighlightedSlice = (event: DataHoverClearEvent) => {
+      if (eventBus.isOwnEvent(event)) {
+        setHighlightedTitle(undefined);
+      }
+    };
+
+    const subs = new Subscription();
+    subs.add(eventBus.subscribe(DataHoverEvent, setHighlightedSlice));
+    subs.add(eventBus.subscribe(DataHoverClearEvent, resetHighlightedSlice));
+
+    return () => {
+      subs.unsubscribe();
+    };
+  }, [setHighlightedTitle, eventBus]);
+
+  return highlightedTitle;
+}
+
```

Comment:
Great refactoring, I don't think it should matter but maybe the code will be clearer if we move the arrow functions outside of the `useEffect`?
```suggestion
function useSliceHighlightState() {
  const [highlightedTitle, setHighlightedTitle] = useState<string>();
  const { eventBus } = usePanelContext();
  const setHighlightedSlice = useCallback(
    (event: DataHoverEvent) => {
      if (eventBus.isOwnEvent(event)) {
        setHighlightedTitle(event.payload.dataId);
      }
    },
    [eventBus]
  );
  const resetHighlightedSlice = useCallback(
    (event: DataHoverClearEvent) => {
      if (eventBus.isOwnEvent(event)) {
        setHighlightedTitle(undefined);
      }
    },
    [eventBus]
  );

  useEffect(() => {
    if (!eventBus) {
      return;
    }

    const subs = new Subscription();
    subs.add(eventBus.subscribe(DataHoverEvent, setHighlightedSlice));
    subs.add(eventBus.subscribe(DataHoverClearEvent, resetHighlightedSlice));

    return () => {
      subs.unsubscribe();
    };
  }, [setHighlightedSlice, resetHighlightedSlice, eventBus]);

  return highlightedTitle;
}
```

Reply:
I agree with Hugo, he's proposal looks cleaner

Reply:
I tested this but means two additional dependencies to the useEffect, which makes the code more complex and the function scoping more complex